tensorflow_lite_support/ios/task/vision/sources/TFLObjectDetector.h

```diff
@@ -49,7 +49,7 @@ NS_ASSUME_NONNULL_BEGIN
  * @return An instance of TFLObjectDetectorOptions set to the specified
  * modelPath.
  */
-- (nullable instancetype)initWithModelPath:(nonnull NSString *)modelPath;
+- (instancetype)initWithModelPath:(NSString *)modelPath;
```

Comment:
We should still mark modelPath as nonnull here, and the `(nullable instancetype)objectDetectorWithOptions:(nonnull TFLObjectDetectorOptions *)options` below. ObjC doesn't enforce the null check at compile time and that's fine. If an error is created and the objectDetectorWithOptions:error: returns nil, the user can examine the error there.

Reply:
modelPath is automatically considered nonnull because of NS_ASSUME_NONNULL_BEGIN in the headers. Only nullable args need to be marked explicitly as nullable.  So compiler will still throw an error if this is null in swift.

Also my doubt is whether to add the nullable to return type here, since modelPath can be null only in ObjC even if it is marked  as nonnull. 
In that case there are 2 options:
 1. Initialise the baseOptions.modelFile.modelPath to null and let TFLObjectDetectorOptions be initialised. In this case return type will be nonnull. 
 2. Check for nullability of modelPath and return a null TFLObjectDetectorOptions. In this case return type will be nullable.  We can add an error to the initialiser as well here.

Reply:
Disregard my comments on the null check. I saw your response.

Reply:
never mind the nonnull comment. I forgot that notnull is the defaut.